src/test/java/com/google/sps/ProfileBuilderServletTest.java

```diff
@@ -0,0 +1,196 @@
+    response = Mockito.mock(HttpServletResponse.class);
+
+    localHelper.setUp();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    localHelper.tearDown();
+  }
+    
+  public void addMentorProfile() {
+    String userType = "Mentor";
+    when(request.getParameter("user-Type")).thenReturn(userType);
+    String mentorName = "Jeff";
+    when(request.getParameter("profName-input")).thenReturn(mentorName);
+    String company = "Google";
+    when(request.getParameter("company-input")).thenReturn(company);
+    String specialty = "Security";
+    when(request.getParameter("specialty-input")).thenReturn(specialty);
+  }
+
+  public void addMenteeProfile() {
+    String userType = "Mentee";
+    when(request.getParameter("user-Type")).thenReturn(userType);
+    String menteeName = "Bobby";
+    when(request.getParameter("name-input")).thenReturn(menteeName);
+    String school = "University of California Berkeley";
+    when(request.getParameter("school-input")).thenReturn(school);
+    String major = "Computer Science";
+    when(request.getParameter("major-input")).thenReturn(major);
+  }
+
+  @Test
+  public void doPostRedirectsUrl() throws IOException, ServletException {
+    addMentorProfile();
+    userServlet.doPost(request, response);
+
+    verify(response, times(1)).sendRedirect("/logged_in_homepage.html");
+   }
+  
+  @Test
+  public void addMentorProfileThenCheckSize() throws IOException, ServletException {
+    DatastoreService dataService = DatastoreServiceFactory.getDatastoreService();
+    addMentorProfile();
+    userServlet.doPost(request, response);
+
+    Assert.assertEquals(1, dataService.prepare(new Query("User")).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void addMenteeProfileThenCheckSize() throws IOException, ServletException {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    addMenteeProfile();
+    userServlet.doPost(request, response);
+
+    Assert.assertEquals(1, ds.prepare(new Query("User")).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void checkMenteeSchoolInfoIsSame() throws IOException, ServletException {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    String userType = "Mentee";
+    when(request.getParameter("user-Type")).thenReturn(userType);
+    String menteeName = "Sandy Cane";
+    when(request.getParameter("name-input")).thenReturn(menteeName);
+    String school = "Stanford University";
+    when(request.getParameter("school-input")).thenReturn(school);
+    String major = "Computer Science";
+    when(request.getParameter("major-input")).thenReturn(major);
+
+    userServlet.doPost(request, response);
+    Filter schoolFilter = new FilterPredicate("school", FilterOperator.EQUAL, school);
+    PreparedQuery results = ds.prepare(new Query("User").setFilter(schoolFilter));
+    String result = "";
+    for (Entity entity : results.asIterable()) {
+        result = (String) entity.getProperty("school");
```

Comment:
rather than checking the fields in separate tests, consider verifying that the entity that is returned from datastore matches the entity you expect to be added based on your request parameters.

Reply:
So, you'd like to have one test check that the entity being return matches the parameters? All in one test?

Reply:
Yes I am suggesting that you check multiple fields in the values from the datastore. If you set ten properties in your request, then you can verify that the ten properties exist in the datastore. 

The principle of separating tests so that each test verifies a single behavior causes your tests to be easier for others to understand. However to test a single behavior, you may need to validate that multiple conditions are true. Because we are testing whether a POST request will save an entity into the datastore, we should also check that all the fields of the entity are correct.

Let me know if that helps.

Reply:
Yea that helps. I made a test to fetch from Datastore, get all the entity properties then compare to whats originally posted

Reply:
Done